src/uu/du/src/du.rs

```diff
@@ -336,14 +336,17 @@ fn du(
                             }
 
                             if let Some(inode) = this_stat.inode {
-                                if seen_inodes.contains(&inode) {
+                                if seen_inodes.contains(&inode)
+                                    && !(options.all && options.count_links)
+                                {
```

Comment:
I don't really understand to be honest. In what cases should this condition be true? For example, I'm not sure how `--all` fits into this, it seems to me like that would be an orthogonal feature?

Reply:
The condition is true if it is a hard link and not both `--all` and `--count-links` were specified by the user. The way `--all` fits in is that it lists the counted files. And by default, hard links are not counted and hence not shown when using `--all`. This changes with `--count-links`, which also counts hard links. And because hard links become "counted files", they must show up with `--all`.

Maybe some example with GNU `du` helps to clarify it:
```
$ mkdir e
$ touch e/f
$ ln e/f e/h
$ ln e/f e/i
$ du --inodes e/
2       e/
$ du --inodes --all e/
1       e/f
2       e/
$ du --inodes --count-links e/
4       e/
$ du --inodes --all --count-links e/
1       e/f
1       e/i
1       e/h
4       e/
```